### PR TITLE
Add new rule: `no-capital-arguments`

### PR DESCRIPTION
**Pull request by lifeart**
resolves: https://github.com/ember-template-lint/ember-template-lint/issues/1732

### CODE REVIEW

File: test/unit/rules/no-capital-arguments-test.js [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+  name: 'no-capital-arguments',
+
+  config: true,
+
+  good: ['<Foo @name="bar" />'],
+
+  bad: [
+    {
+      template: '<Foo @Name="bar" />',
+      fixedTemplate: '<Foo @name="bar" />',
```
Comment (lifeart) [inline test/unit/rules/no-capital-arguments-test.js:17]: fixers should be safe, because capital arguments is not working :)

Comment (rwjblue) [reply test/unit/rules/no-capital-arguments-test.js:16]: No, the fixer is not guaranteed to be safe. The fixer as implemented assumes that the only possible way to use a passed in named argument is via named arguments in the template of the component you are calling (which is not true). For example, `Foo` component in this case could easily be using `this.args.Name` in its JavaScript portion.

File: lib/rules/no-capital-arguments.js [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+const ERROR_MESSAGE = 'Capital argument names is not supported';
+const AllowedPrefix = /[a-z]/;
+
+module.exports = class NoCapitalArguments extends Rule {
+  visitor() {
+    return {
+      AttrNode(node) {
+        if (node.name.startsWith('@')) {
+          let firstChar = node.name.charAt(1);
+          if (!AllowedPrefix.test(firstChar)) {
```
Comment (rwjblue) [inline lib/rules/no-capital-arguments.js:14]: This isn't quite good enough. The current list of disallowed names can be found here:

https://github.com/emberjs/ember.js/blob/3a7c034cdc81c3ed05508f64c630e2041995d2f8/packages/ember-template-compiler/lib/plugins/assert-reserved-named-arguments.ts#L39-L43

```js
const RESERVED = ['@arguments', '@args', '@block', '@else'];

function isReserved(name: string): boolean {
  return RESERVED.indexOf(name) !== -1 || Boolean(name.match(/^@[^a-z]/));
}
```

**Review by rwjblue — CHANGES_REQUESTED** [review]

File: test/unit/rules/no-capital-arguments-test.js (hunk 1 above)
Comment (lifeart) [reply test/unit/rules/no-capital-arguments-test.js:16]: @rwjblue I have added linting for `paths` as well, could we use fixer for paths?

File: lib/rules/no-capital-arguments.js [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+      PathExpression(node) {
+        if (node.data) {
+          let part = node.parts[0] || '';
+          let firstChar = part.charAt(0);
+          let isReserved = this.isReserved(`@${part}`);
+          if (!AllowedPrefix.test(firstChar) || isReserved) {
+            if (this.mode === 'fix' && !isReserved) {
+              node.original = `@${firstChar.toLowerCase()}${part.slice(1)}${node.parts
+                .slice(1)
+                .join('.')}`;
```
Comment (rwjblue) [inline lib/rules/no-capital-arguments.js:28]: I'd rather not have a fixer at all. Changing this to a lowercase value doesn't really fix anything (unless your callsites are all passing the argument with a lowercase as well).

Comment (lifeart) [reply lib/rules/no-capital-arguments.js]: fixers removed

**Review by rwjblue — APPROVED** [review]

**Pull request merged**